usim/py/events.py

```diff
@@ -0,0 +1,642 @@
+                    scope.do(clock('tack'))
+                    scope.do(clock('TOCK'))
+    """
+    __slots__ = '_generator', '_interrupts', 'target'
+
+    def __init__(self, env: 'Environment', generator: Generator[None, Event, V]):
+        if not hasattr(generator, 'send') and not hasattr(generator, 'throw'):
+            raise ValueError(
+                f"'generator' argument must implement 'throw' and 'send'"
+            )
+        super().__init__(env)
+        self._generator = generator
+        self._interrupts = InterruptQueue()
+        #: Event this process is waiting for, or :py:const:`None` if
+        #: the process is starting, stopping or about to be interrupted
+        self.target = None  # type: Optional[Event]
+        env.schedule(self._run_payload(), delay=0)
+
+    def interrupt(self, cause=None):
+        """Interrupt the process by raising an :py:exc:`Interrupt`"""
+        if self._value is None:
+            self._interrupts.push(cause)
+
+    async def _run_payload(self):
+        generator = self._generator
+        interrupts = self._interrupts
+        env = self.env
+        env.active_process = self
+        self.target = event = generator.send(None)  # type: Event
+        env.active_process = None
+        while True:
+            assert isinstance(event, Event),\
+                f'process must yield an Event, not {event.__class__.__name__}'
+            if not event.processed:
+                await (event.__usimpy_flag__ | interrupts.__usimpy_flag__)
+            if interrupts:
+                event = interrupts
+                self.target = None
+            try:
+                if event.ok:
+                    env.active_process = self
+                    self.target = event = generator.send(event.value)
+                    env.active_process = None
+                else:
+                    # the process will handle the exception - or raise a new one
+                    event.defused = True
+                    env.active_process = self
+                    self.target = event = generator.throw(event.value)
+                    env.active_process = None
+            except (StopIteration, StopProcess) as err:
+                value = err.args[0] if err.args else None
+                self.succeed(value)
+                break
+            except BaseException as err:
+                self.fail(err)
+                break
+
+    @property
+    def is_alive(self):
+        """Whether the process is still running"""
+        return self._value is None
+
+    def __repr__(self):
+        return (
+            f'<usim.py.{self.__class__.__name__} object'
+            f' of {self._generator.__name__!r} at {id(self)}>'
+        )
+
+
+class ConditionValue:
+    """
+    ``dict``-like view to the events used in a condition and their value
+
+    .. note::
+
+        This type only captures the events, not their values.
+        If the value of an event changes, this class reflects the change.
```

Comment:
I think the class itself cannot reflect the change but the instance of the class. How correct do you want to be?

Reply:
How about this?

```suggestion
        If the value of a contained event changes, the value in the ConditionValue changes as well.
```